### PR TITLE
PACT-800 Replace integration tests in LogoutControllerSpec with unit tests

### PR DESCRIPTION
**Pull request by felixgb**

### CODE REVIEW

File: test/support/UnitTest.scala [hunk 1]
```diff
@@ -22,12 +22,12 @@
...
 import org.mockito.scalatest.MockitoSugar
-import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach}
+import org.scalatest.{ BeforeAndAfterAll, BeforeAndAfterEach }
 import org.scalatest.matchers.must.Matchers
 import org.scalatestplus.play.PlaySpec
 import play.api.mvc.Results
 
-/**
-  * This trait should be the base class for all unit tests
+/** This trait should be the base class for all unit tests
```
Comment (tna-erasmos) [inline test/support/UnitTest.scala:30]: Really? Even one like `DateRangeSpec`?

File: test/controllers/LogoutControllerSpec.scala [hunk 2]
```diff
@@ -44,21 +44,25 @@ class LogoutControllerSpec extends BaseSpec {
...
         redirectLocation(result) mustBe Some(loginPagePath)
         session(result).get(SessionKeys.token) mustBe empty
-
       }
 
     }
 
   }
 
+  class LogoutTestCase() {
```
Comment (tna-erasmos) [inline test/controllers/LogoutControllerSpec.scala:62]: This might be of general value in the `BaseControllerSpec`.

File: test/controllers/LogoutControllerSpec.scala [hunk 3]
```diff
@@ -44,21 +44,25 @@ class LogoutControllerSpec extends BaseSpec {
         session(result).get(SessionKeys.token) mustBe empty
 
       }
-      "when already logged out" in {
 
-        val controller: LogoutController = inject[LogoutController]
+      "when already logged out" in new LogoutTestCase {
```
Comment (tna-erasmos) [inline test/controllers/LogoutControllerSpec.scala:48]: Why the change in formatting? 

It seems inconsistent both with the style in this class, and elsewhere, where we don't have newlines between these scenario statements.

**Review by tna-erasmos — COMMENTED** [review]
Minor points.

File: test/controllers/LogoutControllerSpec.scala (hunk 3 above)
Comment (felixgb) [reply test/controllers/LogoutControllerSpec.scala:48]: Sorry, I don't understand the question - is it referring to new lines between the test scenarios?

File: test/support/UnitTest.scala (hunk 1 above)
Comment (felixgb) [reply test/support/UnitTest.scala:30]: I did not add this comment, however I do disagree with it as there are plenty of cases where we don't want all of those traits, like you say the date range spec.

Comment (tna-erasmos) [reply test/support/UnitTest.scala:30]: I know; but this is the first time I'm seeing it.

Resolving, as it's out of scope.

File: test/controllers/LogoutControllerSpec.scala (hunk 2 above)
Comment (felixgb) [reply test/controllers/LogoutControllerSpec.scala:62]: I view it as a context for this suite only. I am against adding things to the BaseControllerSpec because:
* I don't want to pollute the namespace for all tests
* I think that mocks will conflict if we set up mocks in the base controller class. Mockito is hard enough to reason about is. I think each test should be responsible for setting up the class that it's testing. In this case, no mocks are being set up, but I imagine they will for the other controllers.

Comment (tna-erasmos) [reply test/controllers/LogoutControllerSpec.scala:62]: OK.

**Review by tna-erasmos — APPROVED** [review]
Only minor formatting issues remain.

File: test/controllers/LogoutControllerSpec.scala (hunk 3 above)
Comment (felixgb) [reply test/controllers/LogoutControllerSpec.scala:48]: I added the newline here because I think that test declarations as similar to function declarations - with a new line before and after the definition.

File: test/support/UnitTest.scala (hunk 1 above)
Comment (felixgb) [reply test/support/UnitTest.scala:30]: I think we should delete this @rwalpole

File: test/support/UnitTest.scala [hunk 4]
```diff
@@ -22,12 +22,12 @@
...
   */
-trait UnitTest extends PlaySpec with Results with Matchers with BeforeAndAfterAll with BeforeAndAfterEach with MockitoSugar
+trait UnitTest
```
[the remark this answers is not in the document]
Comment (rwalpole) [reply test/support/UnitTest.scala:32]: My bad. Thanks @felixgb

File: test/controllers/LogoutControllerSpec.scala (hunk 3 above)
Comment (rwalpole) [reply test/controllers/LogoutControllerSpec.scala:48]: I leave it up to individuals how much white space they put in a file - within reason of course.

**Review by rwalpole — APPROVED** [review]
